airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -1124,18 +1124,17 @@ def build_dataflow_job_name(job_name: str, append_job_name: bool = True) -> str:
 
         return safe_job_name
 
-    @_fallback_to_location_from_variables
     @_fallback_to_project_id_from_variables
     @GoogleBaseHook.fallback_to_default_project_id
     def is_job_dataflow_running(
         self,
         name: str,
         project_id: str,
-        location: str = DEFAULT_DATAFLOW_LOCATION,
+        location: str,
```

Comment:
This is a breaking change.
In the case of version differences between beam and dataflow providers (dataflow newer than beam), this parameter could be provided as `None` - and then an error will be raised.
Personally I don't think that it's harmful to retain the default location, considering the fact that it will be set by the latest beam provider. If we deprecate it, then we should do it more carefully.
Please let me know what you think.

Reply:
I see your point regarding version difference and the potential for resulting incompatibility. I can change back to having the default location if it results in least friction.

It just from my experience using defaults for configuration parameters that should always be provided anyway doesn't lead to anything good. I suppose that wasn't "really" a problem in this case as it seems that no one attempted to use beam/dataflow providers outside US or at least cared to make an issue on it.

Reply:
I understand and agree with your statement - we could deprecate it (now or in another PR), but it should done with some cautious. Something like:
1. Setting the default value to None, rather than `us-central1`.
2. If it's None - use the current `us-central1` value with a deprecation warning that it becomes a mandatory parameter. Otherwise, use the current value.

After it gets released, we could make it mandatory and remove the default `None` value.

Reply:
I've add the deprecation warning as suggested. There is this build step that I can't seem to please (https://github.com/apache/airflow/actions/runs/10991332247/job/30513696655?pr=41887), maybe you could help with what it's complaining about?

Reply:
All checks now passes ✅